pkg/service/system/route53.go

```diff
@@ -129,3 +182,179 @@ func AddRoute53Record(ctx context.Context, dnsName, recordName, regionName strin
+
+	hostedZoneId := config.Get().AwsRoute53HostedZoneID
+
+	// Creating AWS Route53 session
+	route53Client, err := getRoute53Sess(emptyRegion)
+	if err != nil {
+		return nil, errors.Wrap(err, "GetRoute53Records: failed to create route53 session")
+	}
+
+	getRecordsInput := &route53.ListResourceRecordSetsInput{
+		HostedZoneId: aws.String(hostedZoneId),
+		MaxItems:     aws.String("1000"),
+	}
+
+	var records []types.Route53Record
+	var recordSets []*route53.ResourceRecordSet
+
+	// running this loop until we get all the records, getRecordResult.IsTruncated will be false
+	// when we get the last records, route53 returns records in sets of size we mention
+	// in this case we have mentioned value 1000 in input
+	for {
+		getRecordResult, err := route53Client.ListResourceRecordSetsWithContext(ctx, getRecordsInput)
+		if err != nil {
+			return nil, errors.Wrap(err, "GetRoute53Records: failed to get route53 records")
+		}
+
+		recordSets = append(recordSets, getRecordResult.ResourceRecordSets...)
+		if *getRecordResult.IsTruncated {
+			getRecordsInput.StartRecordName = getRecordResult.NextRecordName
+			getRecordsInput.StartRecordType = getRecordResult.NextRecordType
+			getRecordsInput.StartRecordIdentifier = getRecordResult.NextRecordIdentifier
+		} else {
+			break
+		}
+	}
+
+	for _, recordSet := range recordSets {
+		// filtering txt records only
+		if *recordSet.Type != route53.RRTypeTxt {
+			continue
+		}
+
+		for _, resourceRecord := range recordSet.ResourceRecords {
+			record := types.Route53Record{
+				Name:         *recordSet.Name,
+				Value:        *resourceRecord.Value,
+				Type:         *recordSet.Type,
+				TTLInSeconds: int64(*recordSet.TTL),
+			}
+
+			records = append(records, record)
+		}
+	}
+
+	return records, nil
+
+}
+
+// AppendRoute53Records adds records to the zone. It returns the records that were added.
+func AppendRoute53Records(ctx context.Context, records []types.Route53Record) error {
+
+	hostedZoneId := config.Get().AwsRoute53HostedZoneID
+
+	// Creating AWS Route53 session
+	route53Client, err := getRoute53Sess(emptyRegion)
+	if err != nil {
+		return errors.Wrap(err, "AddRoute53Record: failed to create route53 session")
+	}
+
+	hostedZone, err := getHostedZoneFromZoneId(route53Client, hostedZoneId)
+
+	if err != nil {
+		return errors.Wrap(err, "AddRoute53Record: failed to get route53 hostedZone ")
+	}
+
+	createInput := &route53.ChangeResourceRecordSetsInput{
+		ChangeBatch: &route53.ChangeBatch{
+			Changes: make([]*route53.Change, 0, len(records)),
+		},
+		HostedZoneId: aws.String(hostedZoneId),
+	}
+
+	for _, record := range records {
+
+		// AWS Route53 TXT record value must be enclosed in quotation marks on create
+		if record.Type == route53.RRTypeTxt {
+			record.Value = strconv.Quote(record.Value)
+		}
+
+		changeRequest := &route53.Change{
+			Action: aws.String("CREATE"),
+			ResourceRecordSet: &route53.ResourceRecordSet{
+				Name: aws.String(absoluteName(record.Name, hostedZone)),
```

Comment:
Let's not do this, user will have to specify the fqdn as param.

Reply:
fixed